composer/callbacks/image_visualizer.py

```diff
@@ -0,0 +1,189 @@
+            >>> trainer = Trainer(
+            ...     model=model,
+            ...     train_dataloader=train_dataloader,
+            ...     eval_dataloader=eval_dataloader,
+            ...     optimizers=optimizer,
+            ...     max_duration="1ep",
+            ...     callbacks=[ImageVisualizer()],
+            ... )
+
+    The images are logged by the :class:`.Logger` to the following key(s) as described
+    below.
+
+    +---------------------------------------------+---------------------------------------+
+    | Key                                         | Logged data                           |
+    +=============================================+=======================================+
+    | ``Images/Train``                            |   Sampled examples of train images    |
+    +---------------------------------------------+---------------------------------------+
+    | ``Images/Eval``                             |   Sampled examples of eval images     |
+    +---------------------------------------------+---------------------------------------+
+
+    Args:
+        interval (str, optional): Time string specifying how often to log train images. For example, ``interval='1ep'``
+            means images are logged once every epoch, while ``interval='100ba'`` means images are logged once every 100
+            batches. Eval images are logged once at the start of each eval. Default: ``"100ba"``.
+        mode (str, optional): How to log the image labels. Valid values are ``"input"`` (input only)
+            and "segmentation" which also saves segmentation masks. Default: ``"input"``.
+        num_images (int, optional): Number of images to log. Should be less than or equal to than the microbatch size.
+            If there are not enough images in the microbatch, all the images in the microbatch will be logged.
+            Default: ``8``.
+        input_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the input
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair.  The default is 0, which corresponds to any sequence, where the first
+            element is the input. Default: ``0``.
+        target_key (str | int | Tuple[Callable, Callable] | Any, optional): A key that indexes to the target
+            from the batch. Can also be a pair of get and set functions, where the getter
+            is assumed to be first in the pair. The default is 1, which corresponds to any sequence, where the second
+            element is the target. Default: ``1``.
+    """
+
+    def __init__(self,
+                 interval: str = '100ba',
+                 mode: str = 'input',
+                 num_images: int = 8,
+                 input_key: Union[str, int, Tuple[Callable, Callable], Any] = 0,
+                 target_key: Union[str, int, Tuple[Callable, Callable], Any] = 1):
+        self.interval = interval
```

Comment:
nit: I'd remove this line here; then, the `assert isinstance(self.interval, Time)` checks later won't be required.